modules/apps/web-experience/export-import/export-import-resources-importer/src/main/java/com/liferay/exportimport/resources/importer/internal/messaging/ResourcesImporterHotDeployMessageListener.java

```diff
@@ -73,6 +78,16 @@
 
 	@Activate
 	protected void activate(final BundleContext bundleContext) {
+		_bundleContext = bundleContext;
+
+		_messageQueue = new LinkedList<>();
+
+		_serviceTracker = ServiceTrackerFactory.open(
```

Comment:
I see what you mean, actually we found the very same thing, but we were happy that at least it works for now (probably because of pure luck as you said). Unfortunately we couldn't find any better solution to resolve the abstract problem, namely, to somehow make our service wait until all the startup procedure finishes for the available portlets. I am open for any change, to be honest the reason why I sent you the pull was more to start a discussion then to make it into master as it is. Right now in this modularized world for me it seems almost impossible to define a moment in time when we can tell  the portal startup is finished, all available bundles are up an running. Unfortunately if we run the auto importer process before that theoretical moment, the users are going to consider it failed. So my goal would be somehow sync the first import process with the portal startup.